### PR TITLE
GraphScope on 3/3 part 2: pick up the previous graphscope functions

### PR DESCRIPTION
**Pull request by sighingnow**
This pull request picks up necessary routines to support using graphlearn in the context of graphscope (see also the original work in https://github.com/alibaba/graph-learn/commit/e56fc9c8bc92f68eb308b5a91680d7f6abdce57a).

This pull request has done a lot of cleanup and simplification, the main changes there are:

- Add a `Client` class to support connecting to multiple servers from one client
- Add basic routines that need to wrap a vineyard graph to use in graphlearn.

### CODE REVIEW

File: graphlearn/python/client.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+      return False   # the client has only one own server
+    elif next_index < self._cur_index:
+      # one epoch finish, start next epoch
+      self._cur_index = next_index
+      return False
+    else:
+      # one server finish, start to connect next server
+      if self._client_cache[next_index] is None:
+        self._client_cache[next_index] = pywrap.rpc_client(
+          self._own_servers[next_index], True, True
```
Comment (baoleai) [inline graphlearn/python/client.py:80]: There are only two args for `pywrap.rpc_client`

**Review by LiSu — APPROVED** [review]
LGTM

File: graphlearn/python/gsl/dag_dataset.py [hunk 2]
```diff
@@ -54,22 +61,37 @@ def next(self):
...
       if res:
         pywrap.del_get_dag_value_response(res)
       self._cur_res = None
-      raise OutOfRangeError("OutOfRange")
+
+      if self._client.connect_to_next_server():
+        self._current_index = (self._current_index + 1) % len(self._client)
+        # initialize the dataset for the next client
+        if self._dag_datasets[self._current_index] is None:
+          status = self._client.run_dag(self._dag.dag_def, self._current_index != len(self._client) - 1)
```
[the remark this answers is not in the document]
Comment (sighingnow) [reply graphlearn/python/gsl/dag_dataset.py]: Fixed.

File: graphlearn/src/service/request/dag_request.cc [hunk 3]
```diff
@@ -34,6 +34,16 @@ bool DagRequest::ParseFrom(const void* request) {
   return true;
 }
 
+bool DagRequest::ParseFrom(const void* request, const bool copy) {
```
[the remark this answers is not in the document]
Comment (sighingnow) [reply graphlearn/src/service/request/dag_request.cc:37]: That is ok, as discussed.

**Review by Seventeen17 — APPROVED** [review]

**Pull request merged**